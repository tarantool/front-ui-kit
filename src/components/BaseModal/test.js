import React from 'react';
import renderer from 'react-test-renderer';
import { BaseModal } from './index';

jest.mock('react-dom', () => ({
  createPortal: node => node
}));

const action = () => null;

describe('BaseModal', () => {
  it('renders closed', () => {
    const tree = renderer.create(
      <BaseModal visible={false} onClose={action}>Modal content</BaseModal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders opened', () => {
    const tree = renderer.create(
      <BaseModal visible onClose={action}>Modal content</BaseModal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});