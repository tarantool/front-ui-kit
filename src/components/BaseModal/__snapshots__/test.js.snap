// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseModal renders closed 1`] = `null`;

exports[`BaseModal renders opened 1`] = `
<div
  className="css-18615gn"
  onMouseDown={[Function]}
>
  <div
    className="css-1vj4vbw"
    onKeyDown={[Function]}
    tabIndex={0}
  >
    Modal content
    <div
      className="css-alng9s"
      onFocus={[Function]}
      tabIndex="0"
    />
  </div>
</div>
`;
